### PR TITLE
Plot documentation for matrixplot

### PR DESCRIPTION
**Pull request by mbuttner**
Inline documentation for the matrixplot. Addressing issue #1664

### CODE REVIEW

File: scanpy/plotting/_matrixplot.py [hunk 1]
```diff
@@ -63,16 +63,21 @@ class MatrixPlot(BasePlot):
 
     Simple visualization of the average expression of a few genes grouped by
     the category 'bulk_labels'.
+    .. plot::
```
Comment (ivirshup) [inline scanpy/plotting/_matrixplot.py]:
```suggestion
    the category 'bulk_labels'.

    .. plot::
```

File: scanpy/plotting/_matrixplot.py [hunk 2]
```diff
@@ -63,16 +63,21 @@ class MatrixPlot(BasePlot):
...
-    >>> markers = ['C1QA', 'PSAP', 'CD79A', 'CD79B', 'CST3', 'LYZ']
-    >>> sc.pl.MatrixPlot(adata, markers, groupby='bulk_labels').show()
+        import scanpy as sc
+        adata = sc.datasets.pbmc68k_reduced()
+        markers = ['C1QA', 'PSAP', 'CD79A', 'CD79B', 'CST3', 'LYZ']
+        sc.pl.matrixplot(adata, markers, groupby='bulk_labels')
 
     Same visualization but passing var_names as dict, which adds a grouping of
     the genes on top of the image:
+    .. plot::
```
Comment (ivirshup) [inline scanpy/plotting/_matrixplot.py]:
```suggestion
    the genes on top of the image:

    .. plot::
```

File: scanpy/plotting/_tools/__init__.py [hunk 3]
```diff
@@ -808,18 +808,28 @@ def rank_genes_groups_matrixplot(
 
     Examples
     --------
-    >>> import scanpy as sc
-    >>> adata = sc.datasets.pbmc68k_reduced()
-    >>> sc.tl.rank_genes_groups(adata, 'bulk_labels', n_genes=adata.raw.shape[1])
+    .. plot::
```
Comment (ivirshup) [inline scanpy/plotting/_tools/__init__.py:812]:
```suggestion

    .. plot::
```

File: scanpy/plotting/_tools/__init__.py [hunk 4]
```diff
@@ -808,18 +808,28 @@ def rank_genes_groups_matrixplot(
...
+        sc.tl.rank_genes_groups(adata, 'bulk_labels', n_genes=adata.raw.shape[1])
 
     Plot `logfoldchanges` instead of gene expression. In this case a diverging colormap
     like `bwr` or `seismic` works better. To center the colormap in zero, the minimum
     and maximum values to plot are set to -4 and 4 respectively.
     Also, only genes with a log fold change of 3 or more are shown.
-    >>> sc.pl.rank_genes_groups_dotplot(adata,
-    ... n_genes=4, values_to_plot="logfoldchanges", cmap='bwr',
-    ... vmin=-4, vmax=4, min_logfoldchange=3, colorbar_title='log fold change')
+    .. plot::
```
Comment (ivirshup) [inline scanpy/plotting/_tools/__init__.py:824]:
```suggestion

    .. plot::
```

**Review by ivirshup — COMMENTED** [review]
I think you're going to need new lines in front of all of the `.. plot` directives. I believe this will make the doc builds pass.

File: scanpy/plotting/_tools/__init__.py (hunk 4 above)
Comment (mbuttner) [reply scanpy/plotting/_tools/__init__.py:824]: Thanks! I added the blank spaces.

**Comment by giovp** [comment]
looks amazing @mbuttner , really like the example on fine tuning, thank you!

**Review by giovp — APPROVED** [review]

**Pull request merged**